docs/proposal-platforms.md

```diff
@@ -0,0 +1,259 @@
+  platform (job submission, polling, log retrieval, etc) will be randomised. This
+  should help balance the load. If a host becomes unavailable, try another random
+  host from the list, until exhausted. This will handle hosts becoming
+  unavailable after a job is submitted which is a major limitation of the current
+  logic.
+
+* It will be possible to specify the platform for a task as a list. For example,
+  you may have 2 different HPC systems and you are happy for a task to run on
+  either. The list will be randomised and the platforms tried in turn until
+  successful job submission.
+  * Note that the platform will be selected each time a job is run. If you
+    specify a list of platforms, tasks must not assume that retries of that task
+    will use the same platform.
+  * For convenience it will also be possible to specify a platform alias which
+    defines a list of platforms.
+
+* It will be possible to specify the platform as a command in the same way as we
+  can currently specify hosts as a command. The command must return a
+  valid platform (or a list of valid platforms). We hope that this functionality
+  will seldom be needed in the future (the main use at present is to select from
+  a list of login nodes). However, there are still cases where this may be
+  needed. For example running a command which:
+  * Returns the platform which has the shortest queue time.
+  * Chooses the platform according to the day of the week (perhaps you have a
+    platform which can only be used at weekends?).
+
+* Note: it will **not** be possible to specify the platform as an environment
+  variable in the same way as we can currently specify remote hosts as an
+  environment variable (e.g. `platform = $ROSE_ORIG_HOST`).
+  * Unclear why we need this given the Jinja2 variables can be used?
+  * Using `$ROSE_ORIG_HOST` relies on Rose specifying extra options to set this
+    in `[cylc][environment]` which we should avoid if possible.
+
+## Example platform configurations
+
+```ini
+[platforms]
+    [[desktop\d\d|laptop\d\d]]
+        # hosts = platform name (default)
+        # Note: "desktop01" and "desktop02" are both valid and distinct platforms
+    [[sugar]]
+        hosts = localhost
+        batch system = slurm
+    [[hpc]]
+        hosts = hpcl1, hpcl2
+        retrieve job logs = True
+        batch system = pbs
+    [[hpcl1-bg]]
+        hosts = hpcl1
+        retrieve job logs = True
+        batch system = background
+    [[hpcl2-bg]]
+        hosts = hpcl2
+        retrieve job logs = True
+        batch system = background
+[platform aliases]
+    [[hpc-bg]]
+        platforms = hpcl1-bg, hpcl2-bg
+```
+
+Note:
+
+* The default for `hosts` is the platform name and platform section headings can
+  be regular expressions to match multiple platform names. This allows us to
+  define many platforms in a single section and handles the case where you have
```

Comment:
This is effectively the old `rose host-select` functionality of grouping together "like" hosts. Perhaps there is a better way to specify these "groupings"?